code/components/extra-natives-five/src/VehicleExtraNatives.cpp

```diff
@@ -1258,6 +1282,115 @@ static HookFunction initFunction([]()
 		g_overrideCanPedStandOnVehicle = context.GetArgument<bool>(0);
 	});
 
+	// vehicle xenon lights patches to support RGB colors
+	{
+		static struct : jitasm::Frontend
+		{
+			virtual void InternalMain() override
+			{
+				// original code
+				movss(dword_ptr[rbp + 0x1C], xmm2);
+				movss(dword_ptr[rbp + 0x18], xmm1);
+				movss(dword_ptr[rbp + 0x14], xmm0);
+
+				// save registers
+				push(rax);
+				push(rcx);
+				push(rdx);
+				push(r8);
+
+				sub(rsp, 0x30);
```

Comment:
Alignment in `OverrideColor` will be wrong as such. This is risky/wrong in x86-64 ABI.

`0x38` (or `0x28`, as you don't use any of this scratch space yourself) will be fine.